### PR TITLE
warehouse: basic warehouse features

### LINKED ISSUE #2: warehouse: sell products from warehouse
Users should be able to sell products from warehouse. 

For example: 

```
Given 10 boxes of apples 
When 5 boxes are saled 
Then the warehouse should contain 5 boxes of Apples
```

### PR DESCRIPTION
**Pull request by slavidichkov**
This version adds all basic warehouse features by using TDD method.

Fixes #2

### CODE REVIEW

File: production/src/Catalog.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import java.util.Map;
+
+/**
+ * Created by clouway on 15-8-7.
+ */
+public class Catalog {
```
Comment (mgenov) [inline production/src/Catalog.java]: Missing javadoc comment on the catalog class.

File: production/src/Catalog.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import java.util.Map;
+
+/**
+ * Created by clouway on 15-8-7.
```
Comment (mgenov) [inline production/src/Catalog.java]: Your IDE settings is not having correct user template: 

```
@author Miroslav Genov (miroslav.genov@clouway.com)
```

File: production/src/Display.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    public void noNameOrQuantityAndPriceAreInvalid() {
+        text = ("The product has not name, or quantity and price are invalid ");
+    }
+
+    public void canNotToAccept() {
+        text = ("This product can not accepted in this warehouse");
+    }
+
+    public void missingNameOrQuantity() {
+        text = ("Selling error: missing name or quantity");
```
Comment (mgenov) [inline production/src/Display.java]: I don't see why you are using such brackets to surround text ?

File: tests/src/TestAddProduct.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Created by clouway on 15-8-6.
+ */
+public class TestAddProduct {
+    Product cola = new Product("cola", 2.30, 10, 30);
```
Comment (mgenov) [inline tests/src/TestAddProduct.java]: Looks like useless definitions. 

You can map them directly in the map, cause these references are not used from tests: 

```java
.put("cola", new Product("cola",2.30,10,30));
```

File: production/src/Catalog.java (hunk 1 above)
Comment (slavidichkov) [inline production/src/Catalog.java]: done.

File: production/src/Catalog.java (hunk 2 above)
Comment (slavidichkov) [inline production/src/Catalog.java]: done.

File: production/src/Display.java (hunk 3 above)
Comment (slavidichkov) [inline production/src/Display.java]: done.

File: tests/src/TestAddProduct.java (hunk 4 above)
Comment (slavidichkov) [inline tests/src/TestAddProduct.java]: done.

File: productions/src/service/Person.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package service;
+
+/**
+ * Created by clouway on 15-8-25.
+ */
+public class Person {
+    private String age;
```
Comment (mgenov) [inline productions/src/service/Person.java]: Getters could be removed by using immutable fields. 

```java
public final class Person {  
   public final String name;
   public final String age; // why not integer ? 

   public Person(String name, age) {
      this.name = name;
      this.age = age;
   }   
}
```

So this getter stuff will be omitted.

**Comment by mgenov** [comment]
You have to wait for confirmation. 

When reviewers type: LGTM, then you will be able to merge with upstream.

File: productions/src/service/Person.java (hunk 5 above)
Comment (slavidichkov) [inline productions/src/service/Person.java]: by condition must be String "получава годините на човек под формата на String"

**Comment by slavidichkov** [comment]
ok I will know for next time

**Comment by mgenov** [comment]
btw, I saw that you are using the Merge button, please use the only push. 

git push upstream master:slavi -> will push local master to upstream master

**Comment by mgenov** [comment]
Here is documented workflow:

https://github.com/clouway/teamspace/wiki/Git%20Workflow

**Comment by slavidichkov** [comment]
ok

**Pull request merged**